Models/ProductsListPart.cs

```diff
@@ -28,23 +28,27 @@ public class ProductsListPart : ContentPart {
 
         private IEnumerable<ShoppingCartItem> RetrieveItems() {
             var partElement = GetInfosetElement();
-            return partElement.Elements()
-                .Where(el => el.Name.LocalName.StartsWith("ShoppingCartItem_")) //XElements corresponding to items
-                .Select(el =>  //parse the XElement to an item
+            var cartElement = partElement.Element("ShoppingCartItems");
+            if (cartElement == null) {
+                return null;
+            }
+            return cartElement.Elements("ShoppingCartItem")
+                .Select(itemElement =>
                     new ShoppingCartItem(
-                        productId: int.Parse(el.Attribute("ProductId").Value),
-                        quantity: int.Parse(el.Attribute("Quantity").Value),
-                        attributeIdsToValues: el.Elements()
-                            .Where(subel => subel.Name.LocalName.StartsWith("AttributeIdsToValues_")) //XElements corresponding to dictionary entries
-                            .Select(subel => {
-                                var id = int.Parse(subel.Attribute("Id").Value);
-                                var val = new ProductAttributeValueExtended() {
-                                    Value = subel.Attribute("Value")?.Value ?? "",
-                                    ExtendedValue = subel.Attributes("ExtendedValue").Any() ? subel.Attribute("ExtendedValue").Value : null,
-                                    ExtensionProvider = subel.Attributes("ExtensionProvider").Any() ?subel.Attribute("ExtensionProvider").Value : null
-                                };
-                                return new KeyValuePair<int, ProductAttributeValueExtended>(id, val);
-                            }
+                        productId: int.Parse(itemElement.Attribute("ProductId").Value),
```

Comment:
I should have spotted this one earlier: you can actually cast `XElement` to `int`: https://msdn.microsoft.com/en-us/library/bb348319(v=vs.110).aspx